js/customize-object-selector-control.js

```diff
@@ -414,6 +424,20 @@
 			var control = this;
 			control.actuallyEmbed();
 			api.Control.prototype.focus.call( control, args );
+		},
+
+		/**
+		 * Add featured image if select2 data is returning.
+		 *
+		 * @param {object} data post data.
+		 * @returns {string}
+		 */
+		select2Template: function(data){
+			if ( data.hasOwnProperty( 'featured_image' ) && data.featured_image.hasOwnProperty( 'sizes' ) && data.featured_image.sizes.hasOwnProperty( 'thumbnail' ) && data.featured_image.sizes.thumbnail.hasOwnProperty( 'url' ) && ! _.isEmpty( data.featured_image.sizes.thumbnail.url ) ) {
+				return '<span><img width="20" height="20" src="' + data.featured_image.sizes.thumbnail.url + '" style="vertical-align:middle;"> ' + data.text + '</span>';
```

Comment:
Don't use string concatenation for constructing this HTML. See https://vip.wordpress.com/2015/03/25/preventing-xss-in-javascript/

Also, use a stylesheet instead of inline style and width/height attributes.